lib/ShareProvider.js

```diff
@@ -0,0 +1,348 @@
+        // socialapi functionality.  One tweak is that we shoot an event
+        // containing the open graph data.
+        let _dataFn;
+        if (!pageData || sharedURI === gBrowser.currentURI) {
+          messageManager.addMessageListener("PageMetadata:PageDataResult", _dataFn = msg => {
+            messageManager.removeMessageListener("PageMetadata:PageDataResult", _dataFn);
+            let pageData = msg.json;
+            if (graphData) {
+              // overwrite data retreived from page with data given to us as a param
+              for (let p of Object.keys(graphData)) {
+                pageData[p] = graphData[p];
+              }
+            }
+            ShareUtils.sharePage(providerOrigin, pageData, target);
+          });
+          gBrowser.selectedBrowser.messageManager.sendAsyncMessage("PageMetadata:GetPageData", null, {target});
+          return;
+        }
+        // if this is a share of a selected item, get any microformats
+        if (!pageData.microformats && target) {
+          messageManager.addMessageListener("PageMetadata:MicroformatsResult", _dataFn = msg => {
+            messageManager.removeMessageListener("PageMetadata:MicroformatsResult", _dataFn);
+            pageData.microformats = msg.data;
+            ShareUtils.sharePage(providerOrigin, pageData, target);
+          });
+          gBrowser.selectedBrowser.messageManager.sendAsyncMessage("PageMetadata:GetMicroformats", null, {target});
+          return;
+        }
+
+        let provider = Social._getProviderFromOrigin(providerOrigin);
+        if (!provider || !provider.shareURL) {
+          return;
+        }
+
+        let shareEndpoint = OpenGraphBuilder.generateEndpointURL(provider.shareURL, pageData);
+        window.open(shareEndpoint, "share-dialog", "chrome");
+
+        eventTracker.handleUserEvent({
+          event: "SHARE_FROM_TOOLBAR",
+          provider: provider.origin
+        });
+      }
+    }
+  };
+}
+
+// This is to register a ShareUtils object in the browser chrome window and
+// to hide the existing social share (paper airplane) button.
+const Overlay = {
+  init: eventTracker => {
+    for (let win of CustomizableUI.windows) {
+      Overlay.setWindowScripts(win, eventTracker);
+    }
+    Services.obs.addObserver(Overlay, "browser-delayed-startup-finished", false);
+  },
+  uninit: () => {
+    Services.obs.removeObserver(Overlay, "browser-delayed-startup-finished");
+    for (let win of CustomizableUI.windows) {
+      delete win.ShareUtils;
+      if (win.SocialShare.shareButton) {
+        win.SocialShare.shareButton.removeAttribute("hidden");
+      }
+    }
+  },
+  observe: window => {
+    Overlay.setWindowScripts(window);
+  },
+  setWindowScripts: (window, eventTracker) => {
+    Object.defineProperty(window, "ShareUtils", windowProperty(window, eventTracker));
+
+    if (window.SocialShare.shareButton) {
+      window.SocialShare.shareButton.setAttribute("hidden", "true");
```

Comment:
I *think* the answer is it doesn't matter. Just did a quick search and it looks like both are all over mozilla-central: http://searchfox.org/mozilla-central/search?q=setAttribute%28%22hidden%22&path=

You should land our eslint rules with the fixes to mozilla central 😜